test/mysql2_test.rb

```diff
@@ -160,6 +160,26 @@ def test_resource_acquisition_for_query
     end
   end
 
+  def test_semian_allows_rollback
+    client = connect_to_mysql!
+
+    client.query('START TRANSACTION;')
+
+    Semian[:mysql_testing].acquire do
+      client.query('ROLLBACK;')
+    end
+  end
```

Comment:
You haven't opened the circuits and the bulkhead count is not 0 in the tests, so why would this fail if this patch wasn't here?